pkg/daemon/certs/certs.go

```
@@ -0,0 +1,191 @@
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			klog.Errorf("Failed to parse certificate: %v\n", err)
+			continue
+		}
+		certs = append(certs, cert)
+	}
+	return certs, nil
+}
+
+// splitAt is a custom closure function to split data into complete PEM-encoded certificates
+func splitAt(substring string) func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	searchBytes := []byte(substring)
+	searchLen := len(searchBytes)
+	return func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		dataLen := len(data)
+		if atEOF && dataLen == 0 {
+			// Empty file, do nothing
+			return 0, nil, nil
+		}
+
+		// Find next separator and return the token
+		if i := bytes.Index(data, searchBytes); i >= 0 {
+			return i + searchLen, data[0 : i+searchLen], nil
+		}
+
+		if atEOF {
+			// Hit end of file before finding another separator, return rest of the data
+			return dataLen, data, nil
+		}
+		// Otherwise, continue reading file data
+		return 0, nil, nil
+	}
+}
+
+// getAllCerts reads all the certs from the given certificate store
+func getAllCerts(store windows.Handle) ([]*x509.Certificate, error) {
+	var existingSystemCerts []*x509.Certificate
+	var certContext *windows.CertContext
+	var err error
+	for {
+		certContext, err = windows.CertEnumCertificatesInStore(store, certContext)
+		if err != nil {
+			if errors.Is(err, windows.Errno(windows.CRYPT_E_NOT_FOUND)) {
+				// Error code implies we have read all certs:
+				// https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certenumcertificatesinstore
+				break
+			}
+			return nil, fmt.Errorf("Error reading existing system certificate: %v", err)
+		}
+
+		// Convert the certificate bytes to Golang x509.Certificate
+		certBytes := unsafe.Slice(certContext.EncodedCert, certContext.Length)
+		x509Cert, err := x509.ParseCertificate(certBytes)
+		if err != nil {
+			klog.Errorf("Failed to parse certificate from bytes %s: %v", certBytes, err)
```

This implies we are ignoring the error. We need to guarantee that all certs are handled and errors are returned even if one fails.

---

+1, I don't see a reason to not return an error here.